src/coreclr/src/jit/emitxarch.cpp

```diff
@@ -2835,7 +2835,13 @@ void emitter::emitHandleMemOp(GenTreeIndir* indir, instrDesc* id, insFormat fmt,
 
         // Either not generating relocatable code, or addr must be an icon handle, or the
         // constant is zero (which we won't generate a relocation for).
-        assert(!emitComp->opts.compReloc || memBase->IsIconHandle() || memBase->IsIntegralConst(0));
+        //
+        // On x86 we can have an indirection of a non-handle int constant without a reloc.
```

Comment:
Nit: With this change, it's not entirely clear that this comment is an extension of the previous comment. Also, while you're here, the previous comment isn't clear about whether this is a statement (i.e. restating the condition of the assert), or describing a condition, e.g. the if statement below).
Could you reword this as a complete statement? Perhaps:
```
// If we reach here, either:
// - we are not generating relocatable code,
// - the base address is a handle for an integer constant,
// - the base address is a constant zero (which doesn't require a reloc), or
// - the base address is a constant that fits into the memory instruction (this can happen on x86).
//   This last case is captured in the FitsInAddrBase method which is used by Lowering to determine that it can
//   be contained.
```
Of course, please correct if the above is not correct.

Reply:
OK can change the comment.

I didn't like the existing comment, It isn't clear to me why we are testing a different set of invariants here than we when originally set the contained flag.  And actually all 31 bit constants don't require a relocation, not just zero (for both x84 and x86)

Reply:
Maybe I will use you comment and omit this part  _(which doesn't require a reloc)_

Reply:
OK, I changed it to:

```
        // If we reach here, either:
        // - we are not generating relocatable code, (typically the non-AOT JIT case)
        // - the base address is a handle represented by an integer constant,
        // - the base address is a constant zero, or
        // - the base address is a constant that fits into the memory instruction (this can happen on x86).
        //   This last case is captured in the FitsInAddrBase method which is used by Lowering to determine that it can
        //   be contained.
        //
```

Reply:
Great, thanks!